webapi.py

```diff
@@ -0,0 +1,127 @@
+import itertools
+import json
+import uuid
+
+from flask import Blueprint, jsonify, request
+
+from models import Shred, Cluster, ClusterMember
+
+
+app = Blueprint("webapi", __name__)
+
+
+@app.route('/cluster', methods=["GET"])
+@app.route('/cluster/<cluster_id>', methods=["GET"])
+def get_cluster(cluster_id=None):
+    """Renders a single cluster as json.
+
+    If the cluster id is not provided, picks a random one.
+    """
+    batch = request.args.get("batch")
+    if cluster_id is None:
+        cluster = Cluster.get_some(batch=batch)
+    else:
+        cluster = Cluster.objects.get_or_404(pk=cluster_id)
+
+    return jsonify({
+        "success": True,
+        "data": {
+            "cluster": json.loads(cluster.to_json()),
+        },
+    })
+
+
+@app.route('/shred/<shred_id>')
+def get_shred(shred_id):
+    """Looks up a single shred by id."""
+    shred = Shred.objects.get_or_404(id=shred_id)
+    return jsonify({'success': True,
+                    'data': {
+                        'shred': json.loads(shred.to_json())
+                    }})
+
+
+@app.route('/cluster', methods=['POST'])
+def create_cluster():
+    """Merges two clusters.
+
+    Processes a POST request that should contain an object like:
+    { "cluster": {
+        "parents": ["parent1_id", "parent2_id"],
+        "members": [{
+            "shred": "shred1_id",
+            "position": [100, 500],
+            "angle": 35,
+          },
+          ...
+        ]
+    } }
+
+    """
+
+    req = request.get_json().get('cluster')
+
+    parents = map(lambda pk: Cluster.objects.get_or_404(pk=pk), req['parents'])
+
+    if len(parents) != 2:
+        response = jsonify({
+            "success": False,
+            "message": "Wrong number of good parents: %s" % req['parents']})
+        response.status_code = 400
+        return response
+
+    if parents[0].batch != parents[1].batch:
+        response = jsonify({
+            "success": False,
+            "message": "Parents are from different batches: %s %s" % (
+                parents[0].batch, parents[1].batch)})
+        response.status_code = 400
+        return response
+
+    cluster = Cluster(
+            id=str(uuid.uuid1()),
+            users_count=0,
+            users_skipped=[],
+            users_processed=[],
+            batch=parents[0].batch,
+            tags=[],
+            parents=parents,
+            members=[ClusterMember(
+                shred=m['shred'], position=m['position'], angle=m['angle'])
```

Comment:
Shouldn't we check for the presence of this params before diving into objects creation?